libs/libc/Makefile

```diff
@@ -115,7 +115,7 @@ endif
 BINDIR ?= bin
 
 AOBJS = $(patsubst %.S, $(BINDIR)$(DELIM)%$(OBJEXT), $(ASRCS))
-COBJS = $(patsubst %.c, $(BINDIR)$(DELIM)%$(OBJEXT), $(CSRCS))
+COBJS = $(patsubst %.c, $(BINDIR)/%$(OBJEXT), $(CSRCS))
```

Comment:
why not use $(DELIM)

Reply:
#1088 remove the line `DELIM := $(strip /)` in libs/libc/Makefile

`# REVISIT: Backslash causes problems in $(COBJS) target`
`DELIM := $(strip /)`
`BINDIR ?= bin`

`AOBJS = $(patsubst %.S, $(BINDIR)$(DELIM)%$(OBJEXT), $(ASRCS))`
`COBJS = $(patsubst %.c, $(BINDIR)$(DELIM)%$(OBJEXT), $(CSRCS))`

And in Windows native build give an error:
`makefile:132: *** target mode do not include“%”. stop.`

Reply:
look at commit e10ab85d1d39950d719a4bd7d2379c01d8c59a28
**_More changes to restore the native Windows build_** 
@patacongo 
add this two line:
`# REVISIT: Backslash causes problems in $(COBJS) target`
`DELIM := $(strip /)`
I think he knows why?

Reply:
`tools/Config.mk` has:
```
# DELIM - Path segment delimiter character
#
# Depends on this settings defined in board-specific defconfig file installed
# at $(TOPDIR)/.config:
#
#   CONFIG_WINDOWS_NATIVE - Defined for a Windows native build

ifeq ($(CONFIG_WINDOWS_NATIVE),y)
  DELIM ?= $(strip \)
else
  DELIM ?= $(strip /)
endif
```
Isn't this still the case for Windows?

Reply:
> Isn't this still the case for Windows?

Windows is not a GNU environment, in Windows native build, we must use some GNU command(from GnuWin32) to support makefile, some special cases we must use `/` rather than Backslash.
libs/libc/Makefile and mm/Makefile is the special cases,to resolve the  in $(COBJS) target caused by backslash , we must use `/` in $(COBJS) of `libs/libc` and `mm` although In most cause we use `\` in Windows native build.

please look at commit https://github.com/apache/incubator-nuttx/commit/e10ab85d1d39950d719a4bd7d2379c01d8c59a28
**More changes to restore the native Windows build**

gregroy add this two line:
`# REVISIT: Backslash causes problems in $(COBJS) target`
`DELIM := $(strip /)`
in `libc/Makefile` and `mm/Makefile`

it means that use  `/`  in Windows native build as well as in LInux environment,  although In most cause should use `\` in Windows native build.

`DELIM := $(strip /)` was removed from `libc/Makefile` and `mm/Makefile`  in #1088 and result use `\` libs in Windows native build in `libc/Makefile` and `mm/Makefile` ,which broke Windows native build and give an error:
`makefile:132: *** target mode do not include“%”. stop.`